src/stan/model/indexing/rvalue.hpp

```diff
@@ -383,8 +378,7 @@ inline auto rvalue(EigMat&& x,
  * @param[in] depth Depth of indexing dimension.
  * @return Result of indexing matrix.
  */
-template <typename EigMat,
-          stan::internal::require_eigen_dense_dynamic_t<EigMat>* = nullptr>
+template <typename EigMat, require_eigen_dense_dynamic_t<EigMat>* = nullptr>
```

Comment:
There are evals in this function. Is that okay?

Reply:
Yeah so for all of the `min_max` functions we need to evaluate before returning because in one case we would return a `Block<>` and in the other case we would return a block with `Reverse<Block<>>`